src/main/java/io/whitesource/cure/SecureObjectInputStream.java

```diff
@@ -0,0 +1,33 @@
+package io.whitesource.cure;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.*;
+import java.util.*;
+
+public class SecureObjectInputStream extends ObjectInputStream {
+
+    private final Set<String> classesWhitelist;
+    private final Set<String> packagesWhitelist = new HashSet<>();
+
+    public SecureObjectInputStream(InputStream inputStream, Set<String> validClassNames) throws IOException {
+        super(inputStream);
+        this.classesWhitelist = validClassNames;
+    }
+
+    public SecureObjectInputStream(InputStream inputStream, Set<String> validClassNames, Set<String> validPackages) throws IOException {
+        super(inputStream);
+        this.classesWhitelist = validClassNames;
+        this.packagesWhitelist.addAll(validPackages);
+    }
+
+    @Override
+    protected Class<?> resolveClass(ObjectStreamClass desc) throws IOException, ClassNotFoundException {
+        String className = desc.getName();
+        String packageName = StringUtils.substringBeforeLast(className, ".");
+        if (classesWhitelist.contains(className) || packagesWhitelist.contains(packageName)) {
```

Comment:
if package name of class provided is a sub-package of a whitelisted package it will fail as written now